src/diffusers/models/modeling_utils.py

```diff
@@ -729,13 +810,35 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 with accelerate.init_empty_weights():
                     model = cls.from_config(config, **unused_kwargs)
 
+                if hf_quantizer is not None:
+                    hf_quantizer.preprocess_model(
+                        model=model, device_map=device_map, keep_in_fp32_modules=keep_in_fp32_modules
+                    )
+
+                    # We store the original dtype for quantized models as we cannot easily retrieve it
+                    # once the weights have been quantized
+                    # Note that once you have loaded a quantized model, you can't change its dtype so this will
+                    # remain a single source of truth
+                    config["_pre_quantization_dtype"] = torch_dtype
+
                 # if device_map is None, load the state dict and move the params from meta device to the cpu
                 if device_map is None and not is_sharded:
-                    param_device = "cpu"
+                    # `torch.cuda.current_device()` is fine here when `hf_quantizer` is not None.
+                    # It would error out during the `validate_environment()` call above in the absence of cuda.
+                    is_quant_method_bnb = (
+                        getattr(model, "quantization_method", None) == QuantizationMethod.BITS_AND_BYTES
+                    )
+                    if hf_quantizer is None:
+                        param_device = "cpu"
+                    elif is_quant_method_bnb:
+                        param_device = torch.cuda.current_device()
```

Comment:
After the loading refactor, we can safely remove this. This is something that should be modified in `update_device_map`.

Reply:
I think you meant `_determine_device_map()`? Anyway, I have made a comment noting this.